File: backend/scripts/reset_db.py
#!/usr/bin/env python3
"""
Reset script for Tyche Backend database
Clears API keys so you can test fresh key generation

Usage: python scripts/reset_db.py
"""

import os
import sys

# Add parent directory to path
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from dotenv import load_dotenv
load_dotenv()

from db import SessionLocal, engine, Base
from models import APIKey

def reset_database():
    print("🔄 Resetting backend database...")
    
    try:
        with SessionLocal() as db:
            # Count existing keys
            count = db.query(APIKey).count()
            print(f"📊 Found {count} existing API keys")
            
            # Delete all API keys
            db.query(APIKey).delete()
            db.commit()
            print("🗑️  Deleted all API keys")
        
        print("\n✅ Backend database reset complete!")
        print("API key table is now empty.\n")
        
    except Exception as e:
        print(f"❌ Error: {e}")
        sys.exit(1)

if __name__ == "__main__":
    reset_database()

File: tyche-terminal/scripts/reset-db.js
#!/usr/bin/env node
/**
 * Reset script for Tyche Terminal database
 * Clears all users so you can test the signup → plan selection flow
 * 
 * Usage: node scripts/reset-db.js
 */

const mongoose = require('mongoose');
require('dotenv').config({ path: '.env.local' });

const MONGODB_URI = process.env.MONGODB_URI || 'mongodb://localhost:27017/tyche-terminal';

async function resetDatabase() {
    console.log('🔄 Connecting to MongoDB...');

    try {
        await mongoose.connect(MONGODB_URI);
        console.log('✅ Connected to MongoDB');

        // Drop the users collection
        const db = mongoose.connection.db;

        const collections = await db.listCollections().toArray();
        const collectionNames = collections.map(c => c.name);

        if (collectionNames.includes('users')) {
            await db.dropCollection('users');
            console.log('🗑️  Dropped users collection');
        } else {
            console.log('ℹ️  No users collection to drop');
        }

        console.log('\n✅ Database reset complete!');
        console.log('You can now sign up fresh and test the plan selection flow.\n');

    } catch (error) {
        console.error('❌ Error:', error.message);
        process.exit(1);
    } finally {
        await mongoose.disconnect();
        console.log('👋 Disconnected from MongoDB');
    }
}

resetDatabase();
